src/main/java/org/fcrepo/importexport/common/BagProfile.java

```diff
@@ -367,4 +389,109 @@ public void validateConfig(final BagConfig config) {
             }
         }
     }
+
+    /**
+     * Validates this {@link BagProfile} according to the BagIt Profiles specification found at
+     * https://bagit-profiles.github.io/bagit-profiles-specification/
+     *
+     * This checks the following fields:
+     *
+     * BagIt-Profile-Info
+     * Existence of the Source-Organization, External-Description, Version, BagIt-Profile-Identifier, and
+     * BagIt-Profile-Version fields
+     *
+     * Serialization
+     * Is equal to one of "forbidden", "required", or "optional"
+     *
+     * Accept-Serialization
+     * If serialization has a value of required or optional, at least one value is needed.
+     *
+     * Manifests-Allowed
+     * If specified, the {@link BagProfile#getPayloadDigestAlgorithms()} must be a subset of
+     * {@link BagProfile#getAllowedPayloadAlgorithms()}
+     *
+     * Tag-Manifests-Allowed
+     * If specified, the {@link BagProfile#getTagDigestAlgorithms()} must be a subset of
+     * {@link BagProfile#getAllowedTagAlgorithms()}
+     *
+     * Tag-Files-Allowed
+     * If specified, the {@link BagProfile#getTagFilesRequired()} must be a subset of
+     * {@link BagProfile#getTagFilesAllowed()}. If not specified, all tags must match the '*' glob
+     *
+     * Accept-BagIt-Version
+     * At least one version is required
+     */
+    public void validateProfile() {
```

Comment:
This method is only used in tests. Would it make more sense to move it into the `src/test` directory?
I am reluctant to add test code into functional classes.

..unless you are planning on invoking this method differently in the future.